### PR TITLE
Fixup PR-1056

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/eXist-db/exist/pull/1056

**NOTE**: These changes affect the format of the `/db/system/security/config.xml`. I think the changes are forwards compatible i.e. old backups should continue to work with eXist-db. However newer backups once this PR is applied will likely not work with older versions of eXist-db.

**NOTE**: This PR includes an API change where I have corrected a typo and spelling of an existing function name.

As such this should be merged as part of the work for eXist-db 5.0.

### CODE REVIEW

File: src/org/exist/security/SecurityManager.java [hunk 1]
```diff
@@ -108,10 +112,6 @@
    @Deprecated
    Subject getSubjectBySessionId(String sessionid);
 
-   void addGroup(int id, Group group);
```
Comment (shabanovd) [inline src/org/exist/security/SecurityManager.java]: why is it require methods renaming?

File: test/src/org/exist/security/internal/BackupRestoreSecurityPrincipalsTest.java [hunk 2]
```diff
@@ -0,0 +1,291 @@
+/*
```
Comment (shabanovd) [inline test/src/org/exist/security/internal/BackupRestoreSecurityPrincipalsTest.java]: that was backup test that also check security related staff. why did you move it out of backup related tests package?

File: src/org/exist/security/SecurityManager.java (hunk 1 above)
Comment (adamretter) [reply src/org/exist/security/SecurityManager.java]: Good question! I don't know - it was a change that you made in your original PR - https://github.com/eXist-db/exist/pull/1056/files#diff-1ae708c6e5ca91cc9ddcfa7181910ca4L111

File: test/src/org/exist/security/internal/BackupRestoreSecurityPrincipalsTest.java (hunk 2 above)
Comment (adamretter) [reply test/src/org/exist/security/internal/BackupRestoreSecurityPrincipalsTest.java]: Good question! Again, I don't know - as it was a change that you made in your original PR - https://github.com/eXist-db/exist/pull/1056/files#diff-14915bb0f81a3900e16d841b355f91b9

Although after some consideration, this does now feel like the correct place for it after some.

File: src/org/exist/security/SecurityManager.java (hunk 1 above)
Comment (shabanovd) [reply src/org/exist/security/SecurityManager.java]: it must stay `addGroup`

File: test/src/org/exist/security/internal/BackupRestoreSecurityPrincipalsTest.java (hunk 2 above)
Comment (shabanovd) [reply test/src/org/exist/security/internal/BackupRestoreSecurityPrincipalsTest.java]: right. guess, I had exactly same reasoning that time too

File: src/org/exist/security/SecurityManager.java (hunk 1 above)
Comment (adamretter) [reply src/org/exist/security/SecurityManager.java]: Why?

Comment (shabanovd) [reply src/org/exist/security/SecurityManager.java]: because it's api change

Comment (adamretter) [reply src/org/exist/security/SecurityManager.java]: That's fine. This PR already changed the format of the backup, so I consider that it already has API changes and therefore will go in 5.0.0

File: src/org/exist/security/internal/SecurityManagerImpl.java [hunk 3]
```diff
@@ -403,7 +395,7 @@ public Subject authenticate(final String username, final Object credentials) thr
             }
 
             final Subject subject = sessions.read(db1 -> {
-                final Session session = db1.get(credentials);
+                final Session session = db1.get(credentials.toString());
```
Comment (shabanovd) [inline src/org/exist/security/internal/SecurityManagerImpl.java]: this is not correct because `credentials` can be `byte[]`

File: src/org/exist/security/SecurityManager.java (hunk 1 above)
Comment (shabanovd) [reply src/org/exist/security/SecurityManager.java]: anyway, is it possible to rename it back ... I don't want to deal with another api "renaming"

File: src/org/exist/security/internal/SecurityManagerImpl.java (hunk 3 above)
Comment (shabanovd) [reply src/org/exist/security/internal/SecurityManagerImpl.java]: k

File: src/org/exist/security/AbstractRealm.java [hunk 4]
```diff
@@ -65,7 +63,7 @@
     protected final PrincipalDbByName<Group> groupsByName = new PrincipalDbByName<>();
     
 
-    private SecurityManager sm;
+    private SecurityManagerImpl sm;
```
Comment (dizzzz) [inline src/org/exist/security/AbstractRealm.java]: Why the Imp here and not the interface?

Comment (adamretter) [reply src/org/exist/security/AbstractRealm.java]: That is one of the things I corrected from the original PR. Check the final state of the PR

Comment (dizzzz) [reply src/org/exist/security/AbstractRealm.java]: Yes sorry.....
